### PR TITLE
chore(core): replace commit listener with DDL listener

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Continuation of #3602

Enterprise PR: https://github.com/questdb/questdb-enterprise/pull/100

Also gets rid of `SqlCodeGenerator.reduceTaskPool` to avoid thread safety issues

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 142 / 145 (97.93%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|12|13|92.31%|
|:large_blue_circle:|io/questdb/Bootstrap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerImpl.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/CharSequenceLongHashMap.java|77|77|100.00%|
|:large_blue_circle:|io/questdb/MessageBusImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AbstractOperation.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/CheckWalTransactionsJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriterAPI.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerFactoryImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AlterOperation.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 140 / 143 (97.90%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|12|13|92.31%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerImpl.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/CharSequenceLongHashMap.java|77|77|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AbstractOperation.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/CheckWalTransactionsJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriterAPI.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerFactoryImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AlterOperation.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 140 / 145 (96.55%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|12|13|92.31%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerImpl.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/CharSequenceLongHashMap.java|77|77|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AbstractOperation.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/CheckWalTransactionsJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriterAPI.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerFactoryImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AlterOperation.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 140 / 145 (96.55%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|12|13|92.31%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerImpl.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/CharSequenceLongHashMap.java|77|77|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AbstractOperation.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/CheckWalTransactionsJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriterAPI.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerFactoryImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AlterOperation.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 146 / 151 (96.69%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|12|13|92.31%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerImpl.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/CharSequenceLongHashMap.java|77|77|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AbstractOperation.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/CheckWalTransactionsJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriterAPI.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerFactoryImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AlterOperation.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 167 / 172 (97.09%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|9|10|90.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpEventBuffer.java|12|13|92.31%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|14|15|93.33%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerImpl.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/std/CharSequenceLongHashMap.java|77|77|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AbstractOperation.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/DefaultFactoryProvider.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/CheckWalTransactionsJob.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriterAPI.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/FactoryProviderImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoEngine.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncJitFilteredRecordCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cairo/DdlListenerFactoryImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/ops/AlterOperation.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java|1|1|100.00%|

File: core/src/main/java/io/questdb/std/CharSequenceLongHashMap.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
```
[the remark this answers is not in the document]
Comment (bluestreak01) [reply core/src/main/java/io/questdb/std/CharSequenceLongHashMap.java:2]: likewise column permissions are also case sensitive

**Review by bluestreak01 — APPROVED** [review]

File: core/src/main/java/io/questdb/std/CharSequenceLongHashMap.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply core/src/main/java/io/questdb/std/CharSequenceLongHashMap.java:2]: Are these comments still relevant?

**Pull request merged**